metricbeat/module/kubernetes/event/event.go

```diff
@@ -97,17 +99,20 @@ func (m *MetricSet) Run(reporter mb.PushReporter) {
 	now := time.Now()
 	handler := kubernetes.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			reporter.Event(generateMapStrFromEvent(obj.(*kubernetes.Event), m.dedotConfig))
+			reporter.Event(generateMapStrFromEvent(obj.(*kubernetes.Event), m.dedotConfig, m.Logger()))
```

Comment:
Hmm. You should do what it suggests. Update `mb.PushReporter` with `mb.PushReporterV2`.

Later on in 
```
AddFunc: func(obj interface{}) {
 			reporter.Event(generateMapStrFromEvent(obj.(*kubernetes.Event), m.dedotConfig, m.Logger()))
```

you could do something like this
```
AddFunc: func(obj interface{}) {
			event := generateMapStrFromEvent(obj.(*kubernetes.Event), m.dedotConfig)
			reporter.Event(mb.TransformMapStrToEvent("kubernetes", event, nil))
		},
```
We already do something similar in https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_storageclass/state_storageclass.go#L96

@ChrsMark wdyt?

Reply:
Yes `PushReporterV2` is the newest reporter that we should use.